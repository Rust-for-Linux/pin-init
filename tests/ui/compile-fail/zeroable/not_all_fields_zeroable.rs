extern crate pinned_init;
use pinned_init::*;

#[derive(Zeroable)]
struct Foo {
    a: usize,
    b: &'static Foo,
}

fn main() {}
