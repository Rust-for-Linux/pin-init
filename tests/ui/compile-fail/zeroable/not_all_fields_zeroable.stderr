error[E0277]: the trait bound `&'static Foo: pinned_init::Zeroable` is not satisfied
 --> tests/ui/compile-fail/zeroable/not_all_fields_zeroable.rs:7:8
  |
7 |     b: &'static Foo,
  |        ^^^^^^^^^^^^ the trait `pinned_init::Zeroable` is not implemented for `&'static Foo`
  |
note: required by a bound in `assert_zeroable`
 --> tests/ui/compile-fail/zeroable/not_all_fields_zeroable.rs:4:10
  |
4 | #[derive(Zeroable)]
  |          ^^^^^^^^ required by this bound in `assert_zeroable`
  = note: this error originates in the macro `::pinned_init::__derive_zeroable` which comes from the expansion of the derive macro `Zeroable` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider removing the leading `&`-reference
  |
7 -     b: &'static Foo,
7 +     b: Foo,
  |
